### PR TITLE
Add HousingType Class

### PR DESCRIPTION
**Pull request by redcolorbicycle**
Addition of the HousingType class is necessary to represent the housingtype attribute for Buyer and Seller objects.

Implemented the HousingType class to encapsulate the housing type information for buyers and sellers, allowing for better organization and management of buyer and seller creation as house type management.

Using a separate class for housing type ensures separation of concerns and follows the single responsibility principle, improving code readability and maintainability.

This commit also includes necessary adjustments to integrate the Housing Type class with the Buyer, Seller and relevant test classes.

### CODE REVIEW

File: src/main/java/seedu/address/model/house/HousingType.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+/**
+ * Represents a House's housing type.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLevel(String)}
+ */
+public class HousingType {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "HousingType should only be Landed or NonLanded";
```
Comment (lokidoki102) [inline src/main/java/seedu/address/model/house/HousingType.java:17]: I think the message here should be something along the line of `should only be HDB, Condominium, Landed`.

**Review by lokidoki102 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/house/HousingType.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+
+/**
+ * Represents a House's housing type.
+ * Guarantees: immutable; is valid as declared in {@link #isValidLevel(String)}
+ */
+public class HousingType {
```
Comment (zengzihui) [inline src/main/java/seedu/address/model/house/HousingType.java:14]: The naming of this need to follow the new class diagram, which is "InterestedHousingType"

File: src/main/java/seedu/address/model/house/HousingType.java (hunk 1 above)
Comment (zengzihui) [reply src/main/java/seedu/address/model/house/HousingType.java:17]: yup, agreed

**Review by KhoonSun47 — APPROVED** [review]
LGTM!

**Pull request merged**